numba_typing/type_annotations.py

```diff
@@ -21,11 +24,20 @@ def get_func_annotations(func):
 
 def get_cls_annotations(cls):
     """Get annotations of class attributes."""
-    return get_type_hints(cls)
-
-
-if __name__ == '__main__':
-    def foo(a: int, b: int = 3):
-        pass
-
-    print(get_func_annotations(foo))
+    annotations = get_type_hints(cls)
+    for x in annotations:
+        if get_args_union(annotations[x]):
+            annotations[x] = get_args_union(annotations[x])
+    return annotations, {}
+
+
+def get_args_union(annot):
```

Comment:
Maybe make the function to handle any annotation something like that:
```python
def get_annotation_types(annotation):
   """Get types of passed annotation."""
    try:
        if annotation.__origin__ is Union:
            return annotation.__args__
    except AttributeError:
        pass

    return (annotation,)
```

Reply:
Then it could be expanded if necessary.

Reply:
The only comment is I'd prefer this function to always return list. Even if it would contain single item